### PR TITLE
Add default messages for errors

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
Error values should have a default message if one is not provided. See #1482 for an example.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs [hunk 1]
```diff
@@ -2255,10 +2255,111 @@ public static FormulaValue Error(IRContext irContext, FormulaValue[] args)
...
             return result;
+        }
+
+        private static string GetDefaultErrorMessage(ErrorKind errorKind)
+        {
+            switch (errorKind)
+            {
+                case ErrorKind.None:
+                    // Default message that is shown to users when an error was produced, but with kind = 'none'
+                    return "System no error";
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2271]: >"System no error" [](http://example.com/codeflow?start=27&length=17)

Do we need to localize those messages? Maybe we could populate MessageKey field so they can get translated automatically.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs (hunk 1 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2271]: Good point, we definitely should... but I've seen many other messages from the interpreter hardcoded. Let me check with Mike to see what's our story for the interpreter.

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2271]: Messages in the interpreter are not localized today, we have some work items to fix that. While that assembly doesn't have its own resources file, the messages need to stay hardcoded for now.

**Review by MikeStall — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs [hunk 2]
```diff
@@ -2255,10 +2255,111 @@ public static FormulaValue Error(IRContext irContext, FormulaValue[] args)
...
 
-                result.Add(new ExpressionError { Kind = errorKind, Message = messageField?.Value as string });
+                var message = messageField != null ? messageField.Value : GetDefaultErrorMessage(errorKind);
+                result.Add(new ExpressionError { Kind = errorKind, Message = message });
             }
 
             return result;
+        }
+
+        private static string GetDefaultErrorMessage(ErrorKind errorKind)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2265]: >GetDefaultErrorMessage [](http://example.com/codeflow?start=30&length=22)

Where are these coming from?  (PA?)

If somebody adds to ErrorKind, how do they know to update this? 
Can we add a test that just uses refleection to get each value of ErrorKind, calls this method, and ensures valid (eg, doesn't hit default case at end)

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs (hunk 1 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2271]: Yes, tracked with #548

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs (hunk 2 above)
Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/Library.cs:2265]: Yes, the messages come from Power Apps. Good idea about the test, added one.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**